### ISSUE TITLE
"GitHub Services" are deprecated; Move integrations (such as Travis CI) to Webhooks and "GitHub Apps" by February

### ISSUE THREAD

**Issue by DeeDeeG** [RefugeRestrooms/refugerestrooms#514]
## Context

From https://github.com/RefugeRestrooms/refugerestrooms/settings/installations:

>**Note:** [GitHub Services are being deprecated](https://developer.github.com/changes/2018-04-25-github-services-deprecation/). Please contact your integrator for more information on how to migrate or replace a service with [webhooks](https://developer.github.com/webhooks/) or [GitHub Apps](https://developer.github.com/apps/differences-between-apps/#about-github-apps/).

---

From https://developer.github.com/changes/2018-04-25-github-services-deprecation/:

>Note: We no longer support GitHub Services as of October 1, 2018, and GitHub Services will stop working on January 31, 2019. Please see the [blog post](https://developer.github.com/changes/2018-10-01-denying-new-github-services) for details. You can use the [Replacing GitHub Services guide](https://developer.github.com/v3/guides/replacing-github-services) to help you update your services to webhooks.

---

## Scope / difficulty

We currently have the following as services:

- Travis CI
- IRC

Shouldn't be too hard to move Travis to a GitHub Apps or webhooks-based setup. I'm less familiar with IRC, and it's a lower priority in my book, if we do plan to get that moved over to webhooks or a GitHub App.

## Impact

Keep our integrations working, particularly Travis CI.

## How to actually do this:

For Travis CI, go here:
- https://github.com/marketplace/travis-ci/plan/MDIyOk1hcmtldHBsYWNlTGlzdGluZ1BsYW43MA==#pricing-and-setup

Select the Open Source plan, select "RefugeRestrooms" from the "Acount" dropdown, and click "Install it for free". Then confirm billing info and so on.

**Comment by DeeDeeG** [comment]
If I set this up for Travis CI, the free "bill" of "$0.00 a month" would be billed to my account.

This is obviously not a problem for me, but it might be neater if someone such as @tkwidmer or @mi-wood took care of this and had the billing officially go to their account.

I'm not totally clear how easy it is to switch who is "billed" once this is done, so I'm holding off for now.

**Comment by DeeDeeG** [comment]
For the record: We don't need to update Code Climate, since it's integrated via WebHooks.

(There is a GitHub App for Code Climate, [here](https://github.com/marketplace/code-climate),  but it only covers their paid services.)

**Comment by DeeDeeG** [comment]
I can go ahead and set this up now if nobody minds. cc @tkwidmer, @mi-wood

**Comment by DeeDeeG** [comment]
Will try to switch Travis to the "GitHub Apps" system some time today.

**Comment by DeeDeeG** [comment]
Apparently we need to disable our old integration with Travis (via _travis-ci.**org**_) in order to get the new version working (via _travis-ci.**com**_)

I did reach out to Travis CI support, but their hours are 10 AM - 10 PM Central European Time, and it was just after 10 PM there when I emailed, so we probably won't hear back before tomorrow.

I posted more details on Slack, but that's the gist of it. I'm sure we can get this sorted out by the end of the month, in any case. But that's the update for now.

**Comment by DeeDeeG** [comment]
Travis CI is migrated over for this repo (`RefugeRestrooms/refugerestrooms`), as well as `refugerestrooms-android` and `refugerestrooms-ios`.

(I suppose we will be letting the old IRC integration close down, at least for now. But as far as I can tell, no-one has used our IRC channel lately.)

So this issue is taken care of, and can be closed! Also, thanks @hissingpanda, @CarmenGitIt, and @hkellaway for being in the loop on this process, and ensuring it went smoothly.
